testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/social/InstagramLoginPage.java

```diff
@@ -34,11 +34,11 @@ public class InstagramLoginPage extends AbstractSocialLoginPage {
     @FindBy(name = "password")
     private WebElement passwordInput;
 
-    @FindBy(xpath = "//button[text()='Save Info']")
+    @FindBy(xpath = "//button[text()='Save info']")
```

Comment:
Is there a way we don't have to rely on the actual text labels? For example picking the element ID, if that is less frequent to change

Reply:
For the UI test suite we use the [`data-testid` convention](https://testing-library.com/docs/queries/bytestid/). That ensures that `id` attributes are not accidentally duplicated, which breaks HTML standards.

Reply:
That's when you own the HTML code, but in this case we don't, it comes from Instagram.

Reply:
Ahhh, my bad. Yeah, just gotta roll with what they deliver then.

Reply:
The problem is that the Save info button has some nonsense classes and the only thing that makes sense to make sure, that I'm working with the right button, is the text which kinda defines it. The button looks like this in HTML.

```
<button class=" _acan _acap _acas _aj1- _ap30" type="button">
    Save info
</button>
```

It looks like that it is the only button shown in the page in the login flow step, but I don't know if it's okay to refer to "a button" since there might appear any other button.

Reply:
Best to keep it on the text I think.